test_dylink_pthread_static_data is flaky

It fairly often fails with:

```

======================================================================
FAIL [0.001s]: test_dylink_pthread_static_data_modularize (test_other.other)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.8/unittest/case.py", line 60, in testPartExecutor
    yield
  File "/usr/lib/python3.8/unittest/case.py", line 676, in run
    self._callTestMethod(testMethod)
  File "/usr/lib/python3.8/unittest/case.py", line 633, in _callTestMethod
    method()
  File "/root/project/test/common.py", line 915, in resulting_test
    return func(self, *args)
  File "/root/project/test/common.py", line 366, in decorated
    f(self, *args, **kwargs)
  File "/root/project/test/common.py", line 687, in metafunc
    f(self, *args, **kwargs)
  File "/root/project/test/test_other.py", line 2178, in test_dylink_pthread_static_data
    self.do_runf(
  File "/root/project/test/common.py", line 1943, in do_runf
    return self._build_and_run(filename, expected_output, **kwargs)
  File "/root/project/test/common.py", line 1988, in _build_and_run
    js_output = self.run_js(js_file, engine, args,
  File "/root/project/test/common.py", line 1578, in run_js
    self.fail('JS subprocess failed (%s): %s (expected=%s).  Output:\n%s' % (error.cmd, error.returncode, assert_returncode, ret))
  File "/usr/lib/python3.8/unittest/case.py", line 753, in fail
    raise self.failureException(msg)
AssertionError: JS subprocess failed (/root/emsdk/node/22.16.0_64bit/bin/node --stack-trace-limit=50 --trace-uncaught /tmp/emtest_ltr8uczw/emscripten_test_other_3yqtyx97/main.js): 1 (expected=0).  Output:
123

node:internal/event_target:1101
  process.nextTick(() => { throw err; });
                           ^
ExitStatus {
  name: 'ExitStatus',
  message: 'Program terminated with exit(0)',
  status: 0
}
Thrown at:
    at node:internal/event_target:1101:28
    at processTicksAndRejections (node:internal/process/task_queues:85:11)


Node.js v22.16.0


----------------------------------------------------------------------
Ran 1455 tests in 1532.146s

FAILED (failures=2, skipped=24)
```

--- a/test/test_other.py
+++ b/test/test_other.py
@@ -2146,45 +2146,47 @@ Module['postRun'] = () => {
     #   the memory is zero-initialized only once (and not once per thread).
     # * The global object must have a constructor to make sure that it is
     #   constructed only once (and not once per thread).
-    create_file('side.cpp', r'''
-      struct Data {
-          Data() : value(42) {}
-          int value;
-      } data;
-      int * get_address() {
-          return &data.value;
+    create_file('side.c', r'''
+      int value = 0;
+
+      __attribute__((constructor)) void ctor(void) {
+        value = 42;
+      }
+
+      int* get_address() {
+        return &value;
       }
       ''')
-    self.run_process([
-      EMCC,
-      '-o', 'side.wasm',
-      'side.cpp',
-      '-pthread', '-Wno-experimental',
-      '-sSIDE_MODULE'])
+    self.run_process([EMCC, '-o', 'side.wasm', 'side.c', '-pthread', '-Wno-experimental', '-sSIDE_MODULE'])
 
-    create_file('main.cpp', r'''
+    create_file('main.c', r'''
+      #include <assert.h>
       #include <stdio.h>
-      #include <thread>
-      int * get_address();
-      int main(void) {
-          *get_address() = 123;
-          std::thread([]{
-            printf("%d\n", *get_address());
-          }).join();
-          return 0;
+      #include <pthread.h>
+
+      int* get_address();
+
+      void* thread_main(void* arg) {
+        assert(*get_address() == 123);
+        printf("%d\n", *get_address());
+        return NULL;
+      }
+
+      int main() {
+        assert(*get_address() == 42);
+        *get_address() = 123;
+        pthread_t t;
+        pthread_create(&t, NULL, thread_main, NULL);
+        pthread_join(t, NULL);
+        return 0;
       }
       ''')
 
-    self.do_runf(
-      'main.cpp',
-      '123',
-      emcc_args=[
-        '-pthread', '-Wno-experimental',
-        '-sPROXY_TO_PTHREAD',
-        '-sEXIT_RUNTIME',
-        '-sMAIN_MODULE=2',
-        'side.wasm',
-      ])
+    self.do_runf('main.c', '123', emcc_args=['-pthread', '-Wno-experimental',
+                                             '-sPROXY_TO_PTHREAD',
+                                             '-sEXIT_RUNTIME',
+                                             '-sMAIN_MODULE=2',
+                                             'side.wasm'])
 
   def test_dylink_pthread_warning(self):
     err = self.expect_fail([EMCC, '-Werror', '-sMAIN_MODULE', '-pthread', test_file('hello_world.c')])

